app/gui2/src/components/GraphEditor.vue

```diff
@@ -22,10 +23,12 @@ const EXECUTION_MODES = ['design', 'live']
 
 const mode = ref('design')
 const viewportNode = ref<HTMLElement>()
+const componentBrowser = ref<HTMLElement>()
```

Comment:
I believe that this is not the right type for this ref. It is assignedas `ref="componentBrowser"` on `ComponentBrowser`. That will give you the [instance of `ComponentBrowser` component](https://vuejs.org/guide/essentials/template-refs.html#ref-on-component). Unfortunately this is not checked by typescript, so you have to manually check if your template refs are matching the type.

I'm not sure if you even need it? Looks like its only usage is actually probably invalid.

Reply:
Removed, it was indeed no longer needed.